File: задание 2/main.js
function calculateDrinks(n, m) {
    let totalDrinks = n;
    let iterations = 0;
    let emptyBottles = n;

    let steps = [];

    while (emptyBottles > 1) {
        let newFull = Math.floor(emptyBottles / m);
        if (newFull === 0) {
            break;
        }
        totalDrinks += newFull;
        iterations++;
        emptyBottles = emptyBottles % m + newFull;
        steps.push(`Step ${iterations} - Drank ${newFull} of cans left ${emptyBottles} empty cans`);
    }

    return { totalDrinks, iterations, steps };
}

const n = 15;
const m = 4;

const { totalDrinks, iterations, steps } = calculateDrinks(n, m);
console.log(`Maximum number of cans drunk: ${totalDrinks}`);
console.log(`Number of iterations: ${iterations}`);
steps.forEach(step => console.log(step));

File: задание 2/main.kt
import kotlin.math.floor

fun calculateDrinks(n: Int, m: Int): Triple<Int, Int, List<String>> {
    var totalDrinks = n
    var iterations = 0
    var emptyBottles = n

    val steps = mutableListOf<String>()

    while (emptyBottles > 1) {
        val newFull = emptyBottles / m
        if (newFull == 0) {
            break
        }
        totalDrinks += newFull
        iterations++
        emptyBottles = emptyBottles % m + newFull
        steps.add("Step $iterations - Drank $newFull of cans left $emptyBottles empty cans")
    }

    return Triple(totalDrinks, iterations, steps)
}

fun main() {
    print("Enter the number of full cans: ")
    val n = readLine()!!.toInt()

    print("Enter the number of empty cans needed for exchange: ")
    val m = readLine()!!.toInt()

    val (totalDrinks, iterations, steps) = calculateDrinks(n, m)
    println("Maximum number of cans drunk: $totalDrinks")
    println("Number of iterations: $iterations")
    steps.forEach { step -> println(step) }
}